lang/c/src/avro/logical.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#ifndef AVRO_LOGICAL_H
+#define AVRO_LOGICAL_H
+#ifdef __cplusplus
+extern "C" {
+#define CLOSE_EXTERN }
+#else
+#define CLOSE_EXTERN
+#endif
+
+enum avro_logical_type_t {
+  AVRO_LOGICAL_TYPE_NONE,
+  AVRO_DECIMAL,
+  AVRO_DATE,
+  AVRO_TIME_MILLIS,
+  AVRO_TIME_MICROS,
+  AVRO_TIMESTAMP_MILLIS,
+  AVRO_TIMESTAMP_MICROS,
+  AVRO_DURATION
+};
+typedef enum avro_logical_type_t avro_logical_type_t;
+
+typedef struct {
+  avro_logical_type_t type;
+  int precision;
+  int scale; 
+} avro_logical_schema_t;
```

Comment:
I am not a C developer but shouldn't this struct mention `decimal` somewhere ?

Reply:
There's `AVRO_DECIMAL` on line 29 in that file.

Reply:
But one could use any of the other enum values L30-L35 too. 
There is no sense in `AVRO_TIMESTAMP_MICROS` + `precision` + `scale`.

Reply:
This structure was inspired by the C++ implementation, where precision and scale are only set when DECIMAL logical type is used: https://github.com/spektom/avro/blob/76e55d8775efb02b5248aadf0bd512b137b58443/lang/c%2B%2B/api/LogicalType.hh#L46C5-L49C21